CoolestMovieAPI/CoolestMovieAPI/Services/MovieRepository.cs

```diff
@@ -102,13 +106,145 @@ public async Task<IList<MovieDTO>> GetByActor(string actorName)
                         title = y.movie.MovieTitle,
                         description = y.movie.MovieDescription,
                         cast = new Dictionary<string, ActorDTO>() { { y.role, y } }
-                   });
+                   }).ToListAsync();
+
+            return query;
+        }
+
+        public async Task<IList<Movie>> GetMoviesByYear(int year)
+        {
+            _logger.LogInformation($"Getting movie by year: {year}");
 
-                   
-            return await query2.ToListAsync();
+            var query = await _movieContext.Movies
+                .Where(m => m.MovieReleaseYear == year)
+                .ToListAsync();
 
+            return query;
         }
 
-        
+        public async Task<IList<Movie>> GetMoviesByYearGreaterThan(int year)
+        {
+            _logger.LogInformation($"Getting movie by year greater than: {year}");
+
+            var query = await _movieContext.Movies
+                .Where(m => m.MovieReleaseYear >= year)
+                .ToListAsync();
+
+            return query;
+        }
+
+        public async Task<IList<Movie>> GetMoviesByYearLessThan(int year)
+        {
+            _logger.LogInformation($"Getting movie by year less than: {year}");
+
+            var query = await _movieContext.Movies
+                .Where(m => m.MovieReleaseYear <= year)
+                .ToListAsync();
+
+            return query;
+        }
+
+        public async Task<IList<Movie>> GetMoviesByYearSpan(int year, int maxYear)
+        {
+            _logger.LogInformation($"Getting movie by yearspan: {year} - {maxYear}");
+
+            var query = await _movieContext.Movies
+                .Where(m => (m.MovieReleaseYear >= year && m.MovieReleaseYear <= maxYear))
+                .ToListAsync();
+
+            return query;
+        }
+
+        public async Task<IList<Movie>> GetMoviesByRating(double rating)
+        {
+            _logger.LogInformation($"Getting movie by rating: {rating}");
+
+            var query = await _movieContext.Movies
+                .Where(m => m.MovieRating == rating)
+                .ToListAsync();
+
+            return query;
+        }
+
+        public async Task<IList<Movie>> GetMoviesByRatingGreaterThan(double rating)
+        {
+            _logger.LogInformation($"Getting movie by rating greater than: {rating}");
+
+            var query = await _movieContext.Movies
+                .Where(m => m.MovieRating >= rating)
+                .ToListAsync();
+
+            if (query.Count == 0) throw new Exception($"Movie(s) not found with rating greater than: {rating}");
```

Comment:
Interesting way to handle an exception.

Reply:
This is old, now using the new way inside the controller so that we can return the correct error code (Code 404 not found) in this case